Simplified/MyBooks/NYPLMyBooksDownloadCenter.m

```diff
@@ -319,74 +313,39 @@ - (void)URLSession:(__attribute__((unused)) NSURLSession *)session
-          [[NYPLBookRegistry sharedRegistry]
-           setState:NYPLBookStateDownloadSuccessful forIdentifier:book.identifier];
-          [[NYPLBookRegistry sharedRegistry] save];
-        } else if (moveError) {
-          [self logBookDownloadFailure:book
-                                reason:@"Couldn't move book to final disk location"
-                          downloadTask:downloadTask
-                              metadata:@{
-                                @"moveError": moveError,
-                                @"removeError": removeError.debugDescription ?: @"N/A",
-                                @"tmpSavedFileURL": tmpSavedFileURL ?: @"N/A",
-                                @"finalFileURL": finalFileURL ?: @"N/A",
-                              }];
-        }
+        success = [self moveFileAtURL:tmpSavedFileURL
+                 toDestinationForBook:book
+                      forDownloadTask:downloadTask];
         break;
       }
     }
+  } else {
+    [self logBookDownloadFailure:book
+                          reason:@"Download Error"
+                    downloadTask:downloadTask
+                        metadata:@{@"problemDocument":
+                                     problemDoc.dictionaryValue ?: @"N/A"}];
   }
   
   if (!success) {
     dispatch_async(dispatch_get_main_queue(), ^{
-      if (problemDocument) {
-        if ([downloadTask.response indicatesAuthenticationNeedsRefresh:problemDocument]) {
-          NYPLLOG(@"Invalid credentials problem when downloading a book, present sign in VC");
-          [NYPLAccountSignInViewController
-           requestCredentialsUsingExisting:NO
-           completionHandler:^{
-            [self startDownloadForBook:book];
-          }];
-        } else {
-          NSString *formattedMessage = [NSString stringWithFormat:NSLocalizedString(@"DownloadCouldNotBeCompletedFormat", nil), book.title];
-          UIAlertController *alert = [NYPLAlertUtils
-                                      alertWithTitle:@"DownloadFailed"
-                                      message:formattedMessage];
-          [[NYPLProblemDocumentCacheManager sharedInstance] cacheProblemDocument:problemDocument key:book.identifier];
-          [NYPLAlertUtils setProblemDocumentWithController:alert document:problemDocument append:YES];
-
-          if ([problemDocument.type isEqualToString:NYPLProblemDocument.TypeNoActiveLoan]) {
-            [[NYPLBookRegistry sharedRegistry] removeBookForIdentifier:book.identifier];
-          }
-
-          [NYPLAlertUtils presentFromViewControllerOrNilWithAlertController:alert viewController:nil animated:YES completion:nil];
-        }
-      } else if (needsAuth) {
-        NYPLLOG(@"Present sign in VC");
-        [NYPLAccountSignInViewController
-         requestCredentialsUsingExisting:NO
-         completionHandler:^{
-          [self startDownloadForBook:book];
-        }];
+      BOOL hasCredentials = [NYPLUserAccount.sharedAccount hasCredentials];
```

Comment:
this logic is not equivalent, but all we wanted to do here is 2 things: handle the expired creds error, and alert user in all other cases. The original logic was leaving some error paths unhandled.